### PR TITLE
OSSM-1404: Add ghproxy

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: prow/cluster/ghproxy.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    metadata:
+      labels:
+        app: ghproxy
+    spec:
+      tolerations:
+      - key: "node-role.kubernetes.io/master"
+        operator: "Exists"
+        effect: "NoSchedule"
+      nodeSelector:
+        kubernetes.io/hostname: prow-master-03 # This could be a proper label such as `role: ghproxy`
```
Comment (dgn) [inline prow/cluster/ghproxy.yaml:23]: master 3 is already running the bazel-cache- maybe master 1?

Comment (jwendell) [reply prow/cluster/ghproxy.yaml:23]: is this a problem? A quick `oc describe node` showed that master3 is the most "free" node.

Comment (dgn) [reply prow/cluster/ghproxy.yaml:23]: actually we should probably start running most of prow on the master nodes

**Review by dgn — APPROVED** [review]

**Pull request merged**